src/runtime/managedtype.cs

```diff
@@ -75,5 +135,103 @@ internal static bool IsManagedType(IntPtr ob)
             }
             return false;
         }
+
+        internal static IDictionary<ManagedType, TrackTypes> GetManagedObjects()
+        {
+            return _managedObjs;
+        }
+
+        internal static void ClearTrackedObjects()
+        {
+            _managedObjs.Clear();
+        }
+
+        internal static int PyVisit(IntPtr ob, IntPtr visit, IntPtr arg)
+        {
+            if (ob == IntPtr.Zero)
+            {
+                return 0;
+            }
+            var visitFunc = (Interop.ObjObjFunc)Marshal.GetDelegateForFunctionPointer(visit, typeof(Interop.ObjObjFunc));
+            return visitFunc(ob, arg);
+        }
+
+        /// <summary>
+        /// Wrapper for calling tp_clear
+        /// </summary>
+        internal void CallTypeClear()
+        {
+            if (tpHandle == IntPtr.Zero || pyHandle == IntPtr.Zero)
+            {
+                return;
+            }
+            var clearPtr = Marshal.ReadIntPtr(tpHandle, TypeOffset.tp_clear);
+            if (clearPtr == IntPtr.Zero)
+            {
+                return;
+            }
+            var clearFunc = (Interop.InquiryFunc)Marshal.GetDelegateForFunctionPointer(clearPtr, typeof(Interop.InquiryFunc));
+            // TODO: Handle errors base on return value
```

Comment:
If we raise some exceptions directly it may interrupt the shutdown progress. But these errors maybe not critical if you don't care about them, yet we still need a way to report them to users or developers. 
We should consider adding an internal logger for the library, not only here.

Reply:
I may have missed something, but after looking at the cpython codebase, the return values are not used outside of `gc.get_referrers` and `gc.get_referents`. Otherwise the return value is ignored.

Reply:
You can say CPython is the same, with no particular error handlers for these. Most of the time, they just don't care. Because after all, you can't interrupt the clear progress.